azure-servicebus/src/main/java/com/microsoft/azure/servicebus/Message.java

```diff
@@ -72,33 +64,48 @@ public Message()
 	
 	public Message(String content)
 	{
-		this(content.getBytes(DEFAULT_CHAR_SET));
+		this(new MessageBody(content));
 	}
 	
 	public Message(byte[] content)
 	{
-		this(content, DEFAULT_CONTENT_TYPE);
+		this(new MessageBody(content));
+	}
+	
+	public Message(MessageBody body)
+	{
+		this(body, DEFAULT_CONTENT_TYPE);
 	}
 	
 	public Message(String content, String contentType)
 	{
-		this(content.getBytes(DEFAULT_CHAR_SET), contentType);
+		this(new MessageBody(content), contentType);
```

Comment:
>this(new MessageBody(content), contentType); [](start = 2, length = 44)

I wonder if this will break existing customers even though we have deprecated the older methods.
If the customer has been creating Message("abcd"), then now suddenly with new client there would be a back compat issue as older messages would be in AmqpData but new messages would be in AmqpValue.

Reply:
Resolved. Converting string body from these constructors to binary data for backward compatibility.

---
In reply to: [227630275](https://github.com/Azure/azure-service-bus-java/pull/296#discussion_r227630275) [](ancestors = 227630275)